vispy/app/base.py

```diff
@@ -184,12 +184,13 @@ def _vispy_mouse_move(self, **kwargs):
 
     def _vispy_mouse_release(self, **kwargs):
         # default method for delivering mouse release events to the canvas
-        kwargs.update(self._vispy_mouse_data)
-        ev = self._vispy_canvas.events.mouse_release(**kwargs)
-        if ev.button == self._vispy_mouse_data['press_event'].button:
+
+        kwds.update(self._vispy_mouse_data)
+        ev = self._vispy_canvas.events.mouse_release(**kwds)
+        if self._vispy_mouse_data['press_event'] and  self._vispy_mouse_data['press_event'].button == ev.button:
             self._vispy_mouse_data['press_event'] = None
 
-        self._vispy_mouse_data['buttons'].remove(ev.button)
+        if ev.button in self._vispy_mouse_data['buttons']: self._vispy_mouse_data['buttons'].remove(ev.button)
```

Comment:
There will be PEP8 problems with this (run `make flake` to see errors), and you need to use `kwargs` instead of `kwds`. You'll either need to rebase or merge upstream master to get these changes...